### ISSUE TITLE
request for BF16 model

### ISSUE THREAD

**Issue by Kanna-Sridhar** [ultralytics/ultralytics#24471]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Is a BF16 model available for YOLO?

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Kanna-Sridhar, thank you for your interest in Ultralytics 🚀! This is an automated response to help get things moving, and an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.
If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).
Since this appears to be a feature request about BF16 model availability, it would be very helpful if you could share a bit more context on your intended workflow, target hardware, and whether you're referring to training, export, or inference support 🙂. Since you mentioned you'd be open to a PR, extra implementation details or a proposed approach would also help the team review it faster.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by AffanBinFaisal** [comment]
Hi @Kanna-Sridhar, before we can scope a PR it would help to know what you mean by "BF16 model" - it could mean any of these, and each has a different answer:

1. **Training in BF16** - Ultralytics' AMP path is currently FP16-only. The `autocast` wrapper in [`ultralytics/utils/torch_utils.py`](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/utils/torch_utils.py) does not accept a dtype argument, so you can't switch autocast to `torch.bfloat16` from the CLI / config today. Exposing an `amp_dtype` (or similar) flag would be a real enhancement.

2. **BF16 inference on a `.pt` model** - this already works manually on Ampere/Hopper GPUs:
   ```python
   from ultralytics import YOLO
   import torch
   model = YOLO("yolo11n.pt")
   model.model = model.model.to(torch.bfloat16)
   ```
   Just make sure inputs match the dtype.

3. **BF16 export** - there is no `bf16` flag in [`default.yaml`](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/default.yaml). `half=True` produces FP16. TensorRT 10+ and ONNX both support BF16 as a tensor type, but it's not exposed by `model.export(...)` today.

4. **A pretrained `.pt` released in BF16** - all official weights are released in FP32; users cast at load time as in #2.

If you can share:
- your target hardware (e.g. H100, A100, Jetson, TPU),
- which mode you need this in (train / predict / export), and
- the format you'd want it in (`.pt` autocast, ONNX, TensorRT engine, ...)

...it'll be much easier to either point you at an existing path or scope a PR. Happy to help draft one once the surface is clear.

**Comment by banu4prasad** [comment]
#16261 
#6545

**Comment by Y-T-G** [comment]
You can convert weight to BF16 manually 

```
model = YOLO("yolo26n.pt")
model.model.bfloat16()
```
